ext/standard/tests/file/file_get_contents_file_put_contents_5gb.phpt

```diff
@@ -18,17 +18,27 @@ function get_system_memory(): int|float|false
             return ((int)trim($output[1])) * 1024;
         }
     } else {
-        // Unix/Linux-based memory check
-        $memInfo = @file_get_contents("/proc/meminfo");
-        if ($memInfo) {
-            preg_match('/MemFree:\s+(\d+) kB/', $memInfo, $matches);
-            return $matches[1] * 1024; // Convert to bytes
+        // check if running in docker
```

Comment:
do we need something similar for other tests:
https://github.com/search?q=repo%3Aphp%2Fphp-src%20proc%2Fmeminfo&type=code

maybe the logic should be extracted for re-use ?

Reply:
maybe it would be even worth it to create a new php-src function which works on docker and regular unix?
that way the php-src tests could be simplified and php userland would also benefit

Reply:
Thats both a good idea IMHO, I'll create a separate PR for the other tests, once we have this one settled.